src/webview/components/memory-table.tsx

```diff
@@ -98,103 +100,264 @@ interface MemoryTableProps extends TableRenderOptions, MemoryDisplayConfiguratio
     offset: number;
     count: number;
     fetchMemory(partialOptions?: Partial<DebugProtocol.ReadMemoryArguments>): Promise<void>;
+    isMemoryFetching: boolean;
+}
+
+interface MemoryRowListOptions {
+    numRows: number;
+    wordsPerRow: number;
+    bigWordsPerRow: bigint;
+}
+
+interface MemoryRowData {
+    rowIndex: number;
+    startAddress: bigint;
+    endAddress: bigint;
 }
 
-export class MemoryTable extends React.Component<MemoryTableProps> {
+interface MemoryTableState {
+    memory?: Memory;
+    data: MemoryRowData[];
+    selectedData: Nullable<DataTableCellSelection<MemoryRowData[]>>;
+}
+
+const itemHeightSingleGroupPerRow = 31;
+const heightGroupsPerRowGain = 14;
+
+export class MemoryTable extends React.Component<MemoryTableProps, MemoryTableState> {
+
+    protected datatableRef;
+
+    protected get isShowMoreEnabled(): boolean {
+        return !!this.props.memory?.bytes.length;
+    }
+
+    constructor(props: MemoryTableProps) {
+        super(props);
+
+        this.datatableRef = React.createRef<DataTable<MemoryRowData[]>>();
+        this.initState();
+    }
+
+    protected initState(): void {
+        const memory = this.props.memory;
+        const numRows = memory ? this.createMemoryRowListOptions(memory, this.props).numRows : 1;
+        this.state = {
+            data: Array.from({ length: numRows }),
+            selectedData: null,
+        };
+    }
+
+    public componentDidUpdate(prevProps: Readonly<MemoryTableProps>): void {
+        this.onMemoryChange(this.props);
+        this.onOptionsChange(prevProps, this.props);
+    }
```

Comment:
In general, setting state in a `componentDidUpdate` callback [is discouraged](https://react.dev/reference/react/Component#componentdidupdate-caveats). Can you describe the goal of these checks? Could it be something that we handle with a `shouldComponentUpdate` check? Could we use the techniques described [here](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state) (though that's an ancient post) to avoid derived state?

Reply:
Yes! I was able to avoid the derived states. Now, only the `selection` variable is handled within `componentDidUpdate` to **move** the selected cell to the correct position depending on the memory reference and offset.

Reply:
Yes! I was able to avoid the derived states. Now, only the `selection` variable is reset within `componentDidUpdate`.